pkg/agent/plugin/workloadattestor/systemd/systemd_posix.go

```diff
@@ -105,13 +84,23 @@ func getSystemdUnitInfo(ctx context.Context, pid uint) (unitInfo, error) {
 	}
 
 	obj := conn.Object(systemdDBusInterface, unitPath)
-	return DBusUnitObject{obj}, nil
+
+	id, err := getStringProperty(obj, "Id")
+	if err != nil {
+		return nil, err
+	}
+	fragmentPath, err := getStringProperty(obj, "FragmentPath")
+	if err != nil {
+		return nil, err
+	}
+
+	return &DBusUnitInfo{id, fragmentPath}, nil
```

Comment:
We avoid use of unkeyed fields in struct literals:
```suggestion
	return &DBusUnitInfo{UnitID: id, UnitFragmentPath: fragmentPath}, nil
```